### PR TITLE
Feature/create views signup and login

### PR DESCRIPTION
**Pull request by victoria-kit**
## JIRA

* [python-141](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=933&view=detail&selectedIssue=PYTHON-141)
* [python-142](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=933&view=detail&selectedIssue=PYTHON-142)

## Code reviewers

- [x] @meelros
- [x] @BoroviyOrest
- [ ] @Kardan1123
- [ ] @dimigor
- [x] @PetrushynskyiOleksii
- [ ] @Sasha1152

### Second Level Review

- [ ] @lhalam

## Summary of change

Create views for sign up and login

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by BoroviyOrest — DISMISSED** [review]
All is good

**Review by meelros — DISMISSED** [review]
Everything is good!

File: way_to_home/custom_user/models.py [hunk 1]
```diff
@@ -27,6 +28,30 @@ def to_dict(self):
             'phone_number': self.phone_number
         }
 
+    def update(self, password=None, google_token=None, phone_number=None, is_active=None):
+        """Method for object update."""
+        if password:
+            self.password = password
```
Comment (PetrushynskyiOleksii) [inline way_to_home/custom_user/models.py]:
```suggestion
            self.password = password
```
You need to use `.set_password()` method to set the password properly. In your case, you save password just as string.

File: way_to_home/custom_user/models.py [hunk 2]
```diff
@@ -27,6 +28,30 @@ def to_dict(self):
...
+            return True
+        except ValueError:
+            return False
+
+    @classmethod
+    def get_by_email(cls, email):
+        """Method for returns user by email"""
+        try:
+            return cls.objects.get(email=email)
+        except (ValueError, IntegrityError):
```
Comment (PetrushynskyiOleksii) [inline way_to_home/custom_user/models.py]:
```suggestion
        except (ValueError, IntegrityError):
```
? I think the exception are `DoesNotExist, ValueError`

File: way_to_home/custom_user/views.py [hunk 3]
```diff
@@ -1 +1,65 @@
...
+@require_http_methods(["POST"])
+def log_in(request):
+    """Login of the existing user. Handles post and get requests."""
+
+    data = request.body
+    email = data.get('email').lower().strip()
+    password = data.get('password')
+    user = authenticate(email=email, password=password)
+    if not user:
+        return HttpResponse('received password is not valid', status=400)
```
Comment (PetrushynskyiOleksii) [inline way_to_home/custom_user/views.py]:
```suggestion
        return HttpResponse('received password is not valid', status=400)
```
Only password can be invalid? I think better to return a message such as `invalid credentials`

**Review by PetrushynskyiOleksii — CHANGES_REQUESTED** [review]

File: way_to_home/way_to_home/settings.py [hunk 4]
```diff
@@ -82,7 +84,7 @@
 
 DATABASES = {
     'default': {
-        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'ENGINE': 'django.db.backends.postgresql',
```
Comment (PetrushynskyiOleksii) [inline way_to_home/way_to_home/settings.py]:
```suggestion
        'ENGINE': 'django.db.backends.postgresql',
```
Return `postgresql_psycopg2`

Comment (victoria-kit) [reply way_to_home/way_to_home/settings.py]: The `django.db.backends.postgresql_psycopg2` backend has been renamed to `django.db.backends.postgresql` in Django 1.9.

File: way_to_home/custom_user/models.py (hunk 1 above)
Comment (victoria-kit) [reply way_to_home/custom_user/models.py]: ok, done

File: way_to_home/custom_user/models.py (hunk 2 above)
Comment (victoria-kit) [reply way_to_home/custom_user/models.py]: done

File: way_to_home/custom_user/views.py (hunk 3 above)
Comment (victoria-kit) [reply way_to_home/custom_user/views.py]: ok, done

File: way_to_home/way_to_home/settings.py (hunk 4 above)
Comment (victoria-kit) [reply way_to_home/way_to_home/settings.py]: done

**Review by PetrushynskyiOleksii — APPROVED** [review]

**Review by BoroviyOrest — APPROVED** [review]

**Pull request merged**